### PR TITLE
sortingAndVerificationCommit

### PR DESCRIPTION
**Pull request by DenysStetsenko54321**

### CODE REVIEW

File: src/pages/Inventory.page.js [hunk 1]
```diff
@@ -9,6 +9,23 @@ export class InventoryPage extends BaseSwagLabPage {
...
+
+    async selectSortOption(value) {
+        await this.productSorting.selectOption(value);
+    }
+
+    async assertSortingNames() {
+        return this.inventoryItemName.allTextContents();
+    }
+
+    async assertSortingPrices() {
```
Comment (AlinaZhykharieva) [inline src/pages/Inventory.page.js]: the name of this method should be changed. We can name smth to assert when we check/verifying smth inside of the method. And for a method with assertions: it is not a good idea to keep it in Page Object

File: src/pages/Inventory.page.js [hunk 2]
```diff
@@ -9,6 +9,23 @@ export class InventoryPage extends BaseSwagLabPage {
...
 
+    get inventoryItemName() { return this.page.locator('.inventory_item_name '); }
+
+    get inventoryItemPrice() { return this.page.locator('.inventory_item_price'); }
+
+    async selectSortOption(value) {
+        await this.productSorting.selectOption(value);
+    }
+
+    async assertSortingNames() {
```
Comment (AlinaZhykharieva) [inline src/pages/Inventory.page.js]: the sane as on 24 string

File: src/tests/unit10Tests.spec.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+const { expect } = require('@playwright/test');
+const { test } = require('../fixture');
+
+test.describe('Unit 10', () => {
+    test('Perform and verify sorting on the Inventory page', async ({ loginPage, inventoryPage }) => {
+        await loginPage.navigate();
+        await loginPage.performLogin('standard_user', 'secret_sauce');
```
Comment (AlinaZhykharieva) [inline src/tests/unit10Tests.spec.js]: we could keep standard user creds outside of this module

File: src/tests/unit10Tests.spec.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+        await inventoryPage.selectSortOption('lohi');
+        const itemPriceLOHI = await inventoryPage.assertSortingPrices();
+        const sortedPricesLOHI = [...itemPriceLOHI].sort((a, b) => a - b);
+        expect(itemPriceLOHI).toEqual(sortedPricesLOHI);
+
+        await inventoryPage.selectSortOption('hilo');
+        const itemPricesHILO = await inventoryPage.assertSortingPrices();
+        const sortedPricesHILO = [...itemPricesHILO].sort((a, b) => b - a);
+        expect(itemPricesHILO).toEqual(sortedPricesHILO);
```
Comment (AlinaZhykharieva) [inline src/tests/unit10Tests.spec.js]: This one test checks all sortings: if the first sorting check fails, the test will fail and we will not know if other sortings work correctly. So, I suggest to have one or two tests but parametrized.

File: src/pages/Inventory.page.js [hunk 5]
```diff
@@ -9,6 +9,23 @@ export class InventoryPage extends BaseSwagLabPage {
...
+
+    async selectSortOption(value) {
+        await this.productSorting.selectOption(value);
+    }
+
+    async verifySortingNames() {
+        return this.inventoryItemName.allTextContents();
+    }
+
+    async verifySortingPrices() {
```
Comment (AlinaZhykharieva) [inline src/pages/Inventory.page.js]: is this function verifying smth? better to name it getPricesValues or smth like that

File: src/pages/Inventory.page.js [hunk 6]
```diff
@@ -9,6 +9,23 @@ export class InventoryPage extends BaseSwagLabPage {
...
 
+    get inventoryItemName() { return this.page.locator('.inventory_item_name '); }
+
+    get inventoryItemPrice() { return this.page.locator('.inventory_item_price'); }
+
+    async selectSortOption(value) {
+        await this.productSorting.selectOption(value);
+    }
+
+    async verifySortingNames() {
```
Comment (AlinaZhykharieva) [inline src/pages/Inventory.page.js]: This function gets some information from elements, but not verify it. Could be named for example: getSoretingsNames

File: src/tests/unit10Tests.spec.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+        expect(itemNameAZ).toEqual(sortedNamesAZ);
+    });
+
+    test('Perform and verify ZA sorting on the Inventory page', async ({ inventoryPage }) => {
+        await inventoryPage.productSorting.click();
+        await inventoryPage.selectSortOption('za');
+        const itemNameZA = await inventoryPage.verifySortingNames();
+        const sortedNamesZA = [...itemNameZA].sort().reverse();
+        expect(itemNameZA).toEqual(sortedNamesZA);
+    });
```
Comment (AlinaZhykharieva) [inline src/tests/unit10Tests.spec.js]: we can group and parametrize this 2 tests

File: src/tests/unit10Tests.spec.js [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+        expect(itemPriceLOHI).toEqual(sortedPricesLOHI);
+    });
+
+    test('Perform and verify HILO sorting on the Inventory page', async ({ inventoryPage }) => {
+        await inventoryPage.productSorting.click();
+        await inventoryPage.selectSortOption('hilo');
+        const itemPricesHILO = await inventoryPage.verifySortingPrices();
+        const sortedPricesHILO = [...itemPricesHILO].sort((a, b) => b - a);
+        expect(itemPricesHILO).toEqual(sortedPricesHILO);
+    });
```
Comment (AlinaZhykharieva) [inline src/tests/unit10Tests.spec.js]: And we can group and parametrize this 2 tests

**Comment by AlinaZhykharieva** [comment]
Круто! Можна пофіксити імена методів, дати на ревью, замержити а потім робити параметризовані тести

File: package.json [hunk 9]
```diff
@@ -12,7 +12,8 @@
   "author": "",
   "license": "ISC",
   "dependencies": {
-    "@playwright/test": "^1.37.1"
+    "@playwright/test": "^1.37.1",
```
Comment (EugenePetrik) [inline package.json]: You can update the PW version, the latest is `1.45.0`

**Review by EugenePetrik — APPROVED** [review]
Good job

File: src/tests/unit10Tests.spec.js [hunk 10]
```diff
@@ -0,0 +1,31 @@
+const { expect } = require('@playwright/test');
```
Comment (EugenePetrik) [inline src/tests/unit10Tests.spec.js]: Let's rename the file name, i.e. sortingOnTheInventoryPage.spec.js.
This title would be more appropriate.

**Review by EugenePetrik — APPROVED** [review]
One more comment )

**Pull request merged**